pkg/syncer/metadatasyncer.go

```diff
@@ -3328,3 +3339,60 @@ func availabilityZoneCRDeleted(obj interface{}) {
 
 	// TODO: Stop watching on any host events of this cluster
 }
+
+// startCnsVolumeInfoCRInformer creates and starts an informer for CnsVolumeInfo custom resource.
+func startCnsVolumeInfoCRInformer(ctx context.Context, cfg *restclient.Config,
+	metadataSyncer *metadataSyncInformer) error {
+	log := logger.GetLogger(ctx)
+	// Create an informer for CnsVolumeInfo instances.
+	dynInformer, err := k8s.GetDynamicInformer(ctx, cnsvolumeinfov1alpha1.GroupName,
+		cnsvolumeinfov1alpha1.Version, cnsvolumeinfov1alpha1.CnsVolumeInfoPlural, metav1.NamespaceAll, cfg, true)
+	if err != nil {
+		return logger.LogNewErrorf(log, "failed to create dynamic informer for %s CR. Error: %+v",
+			cnsvolumeinfov1alpha1.CnsVolumeInfoSingular, err)
+	}
+	cnsVolumeInformer := dynInformer.Informer()
+	_, err = cnsVolumeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			cnsVolumeInfoCRUpdated(oldObj, newObj, metadataSyncer)
+		},
+	})
+	if err != nil {
+		return logger.LogNewErrorf(log, "failed to add event handler on informer for %q CR. Error: %v",
+			cnsvolumeinfov1alpha1.CnsVolumeInfoSingular, err)
+	}
+	// Start informer.
+	go func() {
+		log.Infof("Informer to watch on %s CR starting..", cnsvolumeinfov1alpha1.CnsVolumeInfoSingular)
+		cnsVolumeInformer.Run(make(chan struct{}))
+	}()
+	return nil
+}
+
+// cnsVolumeInfoCRUpdated updates the VolumeSnapshot StoragePolicyUsage "used" capacity. This is used to track
+// snapshot aggregated capacity usage per volume basis
+func cnsVolumeInfoCRUpdated(oldObj interface{}, newObj interface{}, metadataSyncer *metadataSyncInformer) {
+	_, log := logger.GetNewContextWithLogger()
+	// Verify both objects received.
+	var (
+		oldCnsVolumeInfoObj cnsvolumeinfov1alpha1.CNSVolumeInfo
+		newCnsVolumeInfoObj cnsvolumeinfov1alpha1.CNSVolumeInfo
+	)
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+		newObj.(*unstructured.Unstructured).Object, &newCnsVolumeInfoObj)
+	if err != nil {
+		log.Errorf("cnsVolumeInfoCRUpdated: failed to cast new object %+v to %s. Error: %+v", newObj,
+			cnsvolumeinfov1alpha1.CnsVolumeInfoSingular, err)
+		return
+	}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(
+		oldObj.(*unstructured.Unstructured).Object, &oldCnsVolumeInfoObj)
+	if err != nil {
+		log.Errorf("cnsVolumeInfoCRUpdated: failed to cast old object %+v to %s. Error: %+v", oldObj,
+			cnsvolumeinfov1alpha1.CnsVolumeInfoSingular, err)
+		return
+	}
+	log.Debugf("cnsVolumeInfoCRUpdated: Old CnsVolumeInfo: +%v, Updated CnsVolumeInfo: +%+v",
+		oldCnsVolumeInfoObj, newCnsVolumeInfoObj)
+	// TODO: add logic to update VolumeSnapshot StoragePolicyUsage.
```

Comment:
When are you planning to add the logic?
What happens if update StoragePolicyUsage fails?

Reply:
It'll be in a different PR. The plan for update StoragePolicyUsage failure is to retry multiple times, if failed even then, we'll let the csifullysync handle it. This is current behaviour as well.